### PR TITLE
Update: Plugin compatibility in dedicated cloud gateways

### PR DESCRIPTION
**Pull request by lena-larionova**
### Description
https://konghq.atlassian.net/browse/DOCU-3699
Adding info about plugin compatibility with dedicated cloud gateways. 

This might not be the complete list of incompatible plugins. There is also currently nothing to link to for the differences on configuration for AWS IAM assumeRole.

### Testing instructions

Preview link:
https://deploy-preview-7216--kongdocs.netlify.app/hub/plugins/compatibility/
https://deploy-preview-7216--kongdocs.netlify.app/konnect/compatibility/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

### CODE REVIEW

File: app/_hub/kong-inc/aws-lambda/_metadata/_index.yml [hunk 1]
```diff
@@ -7,7 +7,9 @@ enterprise: true
 paid: true
 konnect: true
 network_config_opts: All
-notes: --
+notes: |
+  <b>Dedicated Cloud Gateways</b>: If you use the IAM assumeRole functiuonality with this plugin,
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/aws-lambda/_metadata/_index.yml:11]:
```suggestion
  <b>Dedicated Cloud Gateways</b>: If you use the IAM assumeRole functionality with this plugin, 
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM! I just left one spelling correction.

File: app/hub/plugins/compatibility/index.md [hunk 2]
```diff
@@ -29,10 +29,12 @@ with any of the following network configurations:
     nodes, so only control plane nodes require a database
     (available in {{site.ce_product_name}} 2.0 and {{site.ee_product_name}} 2.1 onward).
 
-* **{{site.konnect_short_name}} (Kong-hosted cloud)**: Hybrid deployment. Nodes are split into control plane and
+* **{{site.konnect_short_name}} (Kong-hosted cloud)**: 
+  * **Hybrid**: Nodes are split into control plane and
 data plane roles. Kong provides and hosts the control plane and a database with
 {{site.konnect_product_name}}, and you provide the {{site.base_gateway}} data plane nodes (no databases required).
-
+  * [**Dedicated Cloud Gateways:**](/konnect/dedicated-cloud-gateways/)
```
Comment (Guaris) [inline app/hub/plugins/compatibility/index.md:36]:
```suggestion
  * [**Dedicated Cloud Gateways:**](/konnect/gateway-manager/dedicated-cloud-gateways/) 
```

**Comment by Guaris** [comment]
Changed a link, 404 is expected

**Pull request merged**